source/extensions/filters/http/sxg/filter.cc

```diff
@@ -0,0 +1,497 @@
+#include "source/extensions/filters/http/sxg/filter.h"
+
+#include <cstdlib>
+#include <regex>
+#include <string>
+#include <chrono>
+
+#include <openssl/evp.h>
+#include <openssl/bio.h>
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+
+#include "envoy/server/filter_config.h"
+// #include "pinterest/common/knox/client.h"
+
+#include "absl/strings/escaping.h"
+#include "envoy/stats/scope.h"
+#include "envoy/http/codes.h"
+#include "source/common/common/utility.h"
+#include "source/common/http/headers.h"
+#include "source/common/stats/utility.h"
+
+#include "libsxg.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace SXG {
+
+Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::RequestHeaders>
+    accept_handle(Http::CustomHeaders::get().Accept);
+
+template <class T>
+const std::vector<std::string> initializeHeaderPrefixFilters(const T& filters_proto) {
+  std::vector<std::string> filters;
+  filters.reserve(filters_proto.size());
+
+  for (const auto& filter : filters_proto) {
+    filters.emplace_back(filter);
+  }
+
+  return filters;
+}
+
+FilterConfig::FilterConfig(const envoy::extensions::filters::http::sxg::v3alpha::SXG& proto_config,
+                           TimeSource& time_source, std::shared_ptr<SecretReader> secret_reader,
+                           const std::string& stat_prefix, Stats::Scope& scope)
+    : stats_(generateStats(stat_prefix + "sxg.", scope)),
+      duration_(proto_config.has_duration() ? proto_config.duration().seconds() : 604800UL),
+      cbor_url_(proto_config.cbor_url()), validity_url_(proto_config.validity_url()),
+      mi_record_size_(proto_config.mi_record_size() ? proto_config.mi_record_size() : 4096L),
+      should_encode_sxg_header_((proto_config.should_encode_sxg_header().length() > 0)
```

Comment:
@rgs1 is there a better way to handle this? this is the header that is passed back from downstream (ngapp) to signal that envoy should encode the content. i want to allow configuration for the header, but have a sane default ...